firmware/stm32f0/Core/Src/i2c-rtc.c

```diff
@@ -193,7 +195,7 @@ void DS3231_SetAlarm1Second(uint8_t second){
  */
 void DS3231_SetAlarm1Minute(uint8_t minute){
     uint8_t temp = DS3231_GetRegByte(DS3231_A1_MINUTE) & 0x80;
-    uint8_t a1m2 = temp | (DS3231_EncodeBCD(minute) & 0x3f);
+    uint8_t a1m2 = temp | (DS3231_EncodeBCD(minute) & 0x7f);

```

Comment:
Why does the bitmask need to be changed?

Reply:
The minutes and seconds bytes hold value information from bit 0 to bit 6. The 0x3f mask (0b00111111) clears bit 6 and 7. This is wrong, because if bit 6 is set to 1 to represent higher values of seconds/minutes, the mask will clear this bit to 0 and we will set an incorrect value. If we use a 0x7f mask (0b01111111) this error won't happen.

Ex. Seconds are set to 50 seconds. In BCD, this corresponds to 0101 0000. With the 0x3f mask, this will be changed to 0001 0000, which is 10 seconds. 

![Screenshot 2024-01-31 160757](https://github.com/Matt-Lemcke/digital-dozenal-clock/assets/82623272/bbc793bf-c117-4781-b5a3-af7311b1f352)

Reply:
![image](https://github.com/Matt-Lemcke/digital-dozenal-clock/assets/82623272/98c44411-3590-458d-8898-fd2ef4c8a1e9)

Reply:
Makes sense